components/tikv_util/src/quota_limiter.rs

```diff
@@ -26,11 +26,16 @@ const CPU_LIMITER_REFILL_DURATION: Duration = Duration::from_millis(100);
 // completion time of tasks through restrictions of different metrics.
 #[derive(Debug)]
 pub struct QuotaLimiter {
-    cputime_limiter: Limiter,
-    write_bandwidth_limiter: Limiter,
-    read_bandwidth_limiter: Limiter,
+    foreground_cputime_limiter: Limiter,
```

Comment:
How about this
```rust
struct XXLimiter {
    cpu_time: Limiter,
    write_bandwidth: Limiter,
    read_bandwidth: Limiter,
}

struct QuotaLimiter {
    forground: XXLimiter,
    backgrounf: XXLimiter,
}
```

Reply:
Then lots of `if else` in `consume_sample` can be avoided

Reply:
> Then lots of `if else` in `consume_sample` can be avoided

consume_sample() function is refactored. please take another look.

Reply:
> How about this
> 
> ```rust
> struct XXLimiter {
>     cpu_time: Limiter,
>     write_bandwidth: Limiter,
>     read_bandwidth: Limiter,
> }
> 
> struct QuotaLimiter {
>     forground: XXLimiter,
>     backgrounf: XXLimiter,
> }
> ```

Refactor the limiter access